index.js

```diff
@@ -1448,14 +1447,179 @@ module.exports = {
     });
   },
 
+  /********************
+   * Raw transactions *
+   ********************/
+
+  parseRawTransactionResponse: function (rawTransactionResponse, packaged, address, privateKey, cost, callback) {
+    if (this.debug.broadcast) {
+      console.debug("[ethrpc] sendRawTransaction response:", rawTransactionResponse);
+    }
+    if (!rawTransactionResponse) {
+      if (!isFunction(callback)) return errors.RAW_TRANSACTION_ERROR;
+      return callback(errors.RAW_TRANSACTION_ERROR);
+    }
+    if (rawTransactionResponse.error) {
+      if (rawTransactionResponse.message.indexOf("rlp") > -1) {
+        var err = clone(errors.RLP_ENCODING_ERROR);
+        err.bubble = rawTransactionResponse;
+        err.packaged = packaged;
+        if (!isFunction(callback)) return err;
+        return callback(err);
+      } else if (rawTransactionResponse.message.indexOf("Nonce too low") > -1) {
+        if (this.debug.broadcast || this.debug.nonce) {
+          console.info("[ethrpc] Nonce too low, incrementing:", rawTransactionResponse.message, packaged, this.rawTxMaxNonce);
+        }
+        ++this.rawTxMaxNonce;
+        delete packaged.nonce;
+        return this.setNonceThenSubmitRawTransaction(packaged, address, privateKey, callback);
+      }
+      if (!isFunction(callback)) return rawTransactionResponse;
+      return callback(rawTransactionResponse);
+    }
+
+    // rawTransactionResponse is the txhash if nothing failed immediately
+    // (even if the tx is nulled, still index the hash)
+    this.rawTxs[rawTransactionResponse] = {tx: packaged, cost: abi.unfix(cost, "string")};
+
+    // nonce ok, complete sequence
+    if (!isFunction(callback)) return rawTransactionResponse;
+    callback(rawTransactionResponse);
+  },
+
+  submitRawTransaction: function (packaged, address, privateKey, callback) {
+    for (var rawTxHash in this.rawTxs) {
+      if (!this.rawTxs.hasOwnProperty(rawTxHash)) continue;
+      if (this.rawTxs[rawTxHash].tx.nonce === packaged.nonce && (!this.txs[rawTxHash] || this.txs[rawTxHash].status !== "failed")) {
+        packaged.nonce = abi.hex(this.rawTxMaxNonce + 1);
+        if (this.debug.broadcast || this.debug.nonce) {
+          console.debug("[ethrpc] duplicate nonce, incremented:", parseInt(packaged.nonce, 16), this.rawTxMaxNonce);
+        }
+        break;
+      }
+    }
+    if (parseInt(packaged.nonce, 16) <= this.rawTxMaxNonce) {
```

Comment:
It can't be passed to sendRawTransaction with a number.  All that gets passed to sendRawTransaction is the signed hex data.  The nonce (and everything else) has to be correctly formatted prior to being signed.  I agree there's too many conversions though.